pkg/agent/params.go

```diff
@@ -94,6 +93,34 @@ func getParameters(config *datamodel.NodeBootstrappingConfiguration, generatorCo
 			addValue(parametersMap, fmt.Sprintf("%sosImageSKU", agentProfile.Name), cloudSpecConfig.OSImageConfig[datamodel.Distro(agentProfile.Distro)].ImageSku)
 			addValue(parametersMap, fmt.Sprintf("%sosImagePublisher", agentProfile.Name), cloudSpecConfig.OSImageConfig[datamodel.Distro(agentProfile.Distro)].ImagePublisher)
 			addValue(parametersMap, fmt.Sprintf("%sosImageVersion", agentProfile.Name), cloudSpecConfig.OSImageConfig[datamodel.Distro(agentProfile.Distro)].ImageVersion)
+		} else {
+			// Set ImageRef if it is not nil and always set the Windows VHD information in WindowsProfile.
+			// ImageRef will be used to generate ARM template for the agent pool if it is set.
+			// Otherwise, the Windows VHD information in WindowsProfile will be used to generate ARM template.
+			// Priority:
+			//   1. ImageRef in agent pool
+			//   2. ImageRef in WindowsProfile
+			//   3. PIR image in WindowsProfile
+			if agentProfile.ImageRef != nil {
+				addValue(parametersMap, fmt.Sprintf("%sosImageName", agentProfile.Name), agentProfile.ImageRef.Name)
+				addValue(parametersMap, fmt.Sprintf("%sosImageResourceGroup", agentProfile.Name), agentProfile.ImageRef.ResourceGroup)
+			} else if properties.WindowsProfile.HasImageRef() {
+				addValue(parametersMap, fmt.Sprintf("%sosImageName", agentProfile.Name), properties.WindowsProfile.ImageRef.Name)
+				addValue(parametersMap, fmt.Sprintf("%sosImageResourceGroup", agentProfile.Name), properties.WindowsProfile.ImageRef.ResourceGroup)
+			}
+			addValue(parametersMap, fmt.Sprintf("%sosImageOffer", agentProfile.Name), properties.WindowsProfile.WindowsOffer)
+			addValue(parametersMap, fmt.Sprintf("%sosImageSKU", agentProfile.Name), properties.WindowsProfile.GetWindowsSku())
+			addValue(parametersMap, fmt.Sprintf("%sosImagePublisher", agentProfile.Name), properties.WindowsProfile.WindowsPublisher)
+			addValue(parametersMap, fmt.Sprintf("%sosImageVersion", agentProfile.Name), properties.WindowsProfile.ImageVersion)
+		}
```

Comment:
actually we don't need any of these data.

Can you also help fix by removing the linux part ? we only care about cse so no vhd version is needed

Reply:
Yeah, I did not find any reference of these `XXosImageXX` in CSE and Customdata, I am going to remove this, what do you think @AbelHu

Reply:
Thanks for Tony's suggestion. I think that we can remove it since these will handled by arm template.